### PR TITLE
App config

### PR DESCRIPTION
**Pull request by coder11**
Still have to update readme

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "eslint.enable": false
+}
```
Comment (DieMyst) [inline .vscode/settings.json:3]: new lines

Comment (coder11) [reply .vscode/settings.json:3]: aah, forgot to remove it

File: src/commands/deployApp.ts [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+}).or('file', 'url');
+
+const scriptsSchema = Joi.object({
+	file: Joi.string(),
+	url: Joi.string().uri(),
+	variables: Joi.object().optional(),
+}).or('file', 'url');
+
+const appConfigSchema = Joi.object({
+	services: Joi.object({}) //
```
[the remark this answers is not in the document]
Comment (coder11) [reply src/commands/deployApp.ts:44]: that's for prettier :) They changed the default behavior for method chaining, so the simplest hack is to add an empty comment)

File: src/commands/deployApp.ts [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+	file: Joi.string(),
+	url: Joi.string().uri(),
+	config: Joi.object({
+		mapped_dirs: Joi.array().items(Joi.string()).optional(),
+		mounted_binaries: Joi.object().optional(),
+		preopened_files: Joi.object().optional(),
+	}),
+});
+
+const scriptStorageSchema = Joi.object({
```
[the remark this answers is not in the document]
Comment (coder11) [reply src/commands/deployApp.ts:30]: Well, there is a lib for generating these types. I dunno, do we need them that badly? I'd prefer a suite of unit tests for config files

**Pull request merged**